conf/script/prov-m0-reset

```diff
@@ -244,137 +249,166 @@ get_fid() {
+    run_cmd $node2 "sudo systemctl start hare-consul-agent-c2"
 
-    while [[ `systemctl is-active hare-consul-agent-c1` != 'active' ||
-             `systemctl is-active hare-consul-agent-c2` != 'active' ]]; do
-           sleep 5
+    while [[ ($(get_leader) != $lnode) && ($(get_leader) != $rnode) ]]; do
+        sleep 1
     done
 
     # An issue is observed if we try to fetch data from Consul immediately
     # after stating Consul services. So, let's wait for 5 more secs.
     sleep 5
 
     echo 'Starting Hax..'
-    sudo systemctl start hare-hax-c1
-    sudo systemctl start hare-hax-c2
+    run_cmd $node1 "sudo systemctl start hare-hax-c1"
+    run_cmd $node2 "sudo systemctl start hare-hax-c2"
 
-    while [[ `systemctl is-active hare-hax-c1` != 'active' ||
-             `systemctl is-active hare-hax-c2` != 'active' ]]; do
+    while [[ $(run_cmd $node1 "systemctl is-active hare-hax-c1") != 'active' ||
+             $(run_cmd $node2 "systemctl is-active hare-hax-c2") != 'active' ]]; do
            sleep 5
     done
 
     echo 'Doing Motr mkfs..'
     m0_mkfs
 
     echo 'Stopping Hax..'
-    sudo systemctl stop hare-hax-c2
-    sudo systemctl stop hare-hax-c1
+    run_cmd $node2 "sudo systemctl stop hare-hax-c2"
+    run_cmd $node1 "sudo systemctl stop hare-hax-c1"
 
     echo 'Stopping Consul agents..'
-    sudo systemctl stop hare-consul-agent-c2
-    sudo systemctl stop hare-consul-agent-c1
+    run_cmd $node2 "sudo systemctl stop hare-consul-agent-c2"
+    run_cmd $node1 "sudo systemctl stop hare-consul-agent-c1"
 
     echo 'Deleting lnet interfaces..'
-    sudo lnetctl net del --net $net_type --if $iface:c1
-    sudo lnetctl net del --net $net_type --if $iface:c2
+    run_cmd $node1 "sudo lnetctl net del --net $net_type --if $iface:c1"
+    run_cmd $node2 "sudo lnetctl net del --net $net_type --if $iface:c2"
 
     echo 'Restarting motr-kernel..'
-    sudo systemctl restart motr-kernel
+    run_cmd $node1 "sudo systemctl restart motr-kernel"
+    run_cmd $node2 "sudo systemctl restart motr-kernel"
 
     echo 'Restoring Consul systemd units..'
-    sudo sed -i '/\(#ExecStartPost\|#ExecStopPost\)=\/usr\/sbin\/attrd_updater/s/#//g' \
-                                  /usr/lib/systemd/system/hare-consul-agent-c1.service
-    sudo sed -i '/\(#ExecStartPost\|#ExecStopPost\)=\/usr\/sbin\/attrd_updater/s/#//g' \
-                                  /usr/lib/systemd/system/hare-consul-agent-c2.service
+    run_cmd $node1 "sudo sed -i '/\(#ExecStartPost\|#ExecStopPost\)=\/usr\/sbin\/attrd_updater/s/#//g' \
+                        /usr/lib/systemd/system/hare-consul-agent-c1.service"
+    run_cmd $node2 "sudo sed -i '/\(#ExecStartPost\|#ExecStopPost\)=\/usr\/sbin\/attrd_updater/s/#//g' \
+                        /usr/lib/systemd/system/hare-consul-agent-c2.service"
 
-    sudo systemctl daemon-reload
-    sudo systemctl reset-failed
+    run_cmd $node1 "sudo systemctl daemon-reload"
+    run_cmd $node2 "sudo systemctl reset-failed"
```

Comment:
these commands need to be run on both servers @mssawant ?

Reply:
@indrajitzagade , now, by default the commands are executed on both the nodes but there's a single node option as well to run the commands from singlenode for both the nodes.

Reply:
systemd reload and reset-failed are not being run on both the nodes.
They need to be run on both the nodes.
​

Reply:
Good catch, let me fix this.